src/FEValues/interface_values.jl

```diff
@@ -0,0 +1,265 @@
+# Defines InterfaceValues and common methods
+"""
+    InterfaceValues(quad_rule::FaceQuadratureRule, func_interpol::Interpolation, [geom_interpol::Interpolation])
+
+An `InterfaceValues` object facilitates the process of evaluating values, averages, jumps and gradients of shape functions
+and nodal functions on the interfaces of finite elements.
+
+**Arguments:**
+
+* `quad_rule`: an instance of a [`FaceQuadratureRule`](@ref)
+* `func_interpol`: an instance of an [`Interpolation`](@ref) used to interpolate the approximated function
+* `geom_interpol`: an optional instance of an [`Interpolation`](@ref) which is used to interpolate the geometry.
+  By default linear Lagrange interpolation is used.
+ 
+**associated methods:**
+
+* [`shape_value_average`](@ref)
+* [`shape_value_jump`](@ref)
+* [`shape_gradient_average`](@ref)
+* [`shape_gradient_jump`](@ref)
+
+**Common methods:**
+
+* [`reinit!`](@ref)
+* [`getnquadpoints`](@ref)
+* [`getdetJdV`](@ref)
+
+* [`shape_value`](@ref)
+* [`shape_gradient`](@ref)
+* [`shape_divergence`](@ref)
+* [`shape_curl`](@ref)
+
+* [`function_value`](@ref)
+* [`function_gradient`](@ref)
+* [`function_symmetric_gradient`](@ref)
+* [`function_divergence`](@ref)
+* [`function_curl`](@ref)
+* [`spatial_coordinate`](@ref)
+"""
+InterfaceValues
+
+struct InterfaceValues{IP, FV<:FaceValues} <: AbstractValues
+    face_values::FV
+    other_face_values::FV
+    # used for quadrature point syncing
+    grid::Grid
+    cell_idx::ScalarWrapper{Int}
+    other_cell_idx::ScalarWrapper{Int}
+    ioi::ScalarWrapper{InterfaceOrientationInfo}
+end
+function InterfaceValues(grid::AbstractGrid, quad_rule::FaceQuadratureRule, func_interpol::Interpolation,
+    geom_interpol::Interpolation = func_interpol)
+    #@assert isDiscontinuous(func_interpol) "`InterfaceValues` is designed for discontinuous interpolations. a continuous interpolation is passed" TODO: add this when sparsity_pattern is merged
+    face_values = FaceValues(quad_rule, func_interpol, geom_interpol)
+    other_face_values = FaceValues(deepcopy(quad_rule), func_interpol, geom_interpol)
+    return InterfaceValues{typeof(func_interpol), FaceValues}(face_values, other_face_values, grid, ScalarWrapper(0), ScalarWrapper(0), ScalarWrapper(InterfaceOrientationInfo(false, nothing)))
+end
+# Maybe move this to common_values.jl?
```

Comment:
I think this is fine here, because the functionality is specific to "interface stuff".